query-engine/connectors/sql-query-connector/src/database/operations/write.rs

```diff
@@ -12,16 +16,82 @@ use std::{
 use tracing::log::trace;
 use user_facing_errors::query_engine::DatabaseConstraint;
 
+async fn generate_id(
```

Comment:
Bool return parameters should always have a doc comment attached explaining what it means if it's not obvious from the signature, which it isn't in this case.

Reply:
Fixed it in a different way, thanks to suggestions from you and Garren I restructured the function into something more "compact" and a single return.